getting-started/optional_installation_settings.md

```diff
@@ -0,0 +1,220 @@
+# Optional Installation Settings
+## Testing Support Setup
+
+Some of our tests we present in the guide examples utilize non-standard requirements that can be appended to your local rails helper. These additions will help provide an error free learning experience! Let's start with our gemfile to make sure we have required dependencies:
+
+{% code title="Gemfile" %}
+```ruby
+group :development, :test do
+  # Preferred testing suite which we use in our Solidus guides
+  gem 'rspec-rails'
+  # Setup 'factories' or prebuilt objects to expedite testing
+  gem 'factory_bot_rails' '~> 4.8'
+  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
+  gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+end
+
+#...
+
+group :test do
+# Adds support for Capybara system testing and selenium driver
+  gem 'capybara', '>= 3.26'
+  # Easy installation and use of web drivers to run system tests with browsers
+  gem 'webdrivers'
+  # Scrubs database clean between tests
+  gem 'database_cleaner-active_record'
+```
+{% endcode %}
+
+Let's go over installation of these gems and why we are suggesting them!
+### RSpec
+RSpec is highly versatile testing framework that we utilize for Solidus and in our guide examples. To install it for your local project, simply Update your bundle and `Gemfile.lock` then install RSpec base files with the following commands in your shell:
+
+```bash
+bundle install
```

Comment:
Shouldn't we first add `rspec` to the `Gemfile`?

Reply:
At the beginning of the guide, I list the Gemfile with all the gems that we will be using. Does that feel too out of place?

Reply:
Sorry, it's ok. It's just that reviewing chunk by chunk makes it difficult to keep the context, but looking at the file, it's clear.